include/AIToolbox/MDP/Algorithms/MCTS.hpp

```diff
@@ -182,8 +180,7 @@ namespace AIToolbox::MDP {
     size_t MCTS<M, Sel, ST, AT>::sampleAction(const ST s, const unsigned horizon) {
```

Comment:
The problem here is that it returns the index of the action. It's a bit counter intuitive to then pull the ```ActionNode``` at this index to then get the ```action``` at this node. And then give this index again to ```sampleAction(a, s, horizon)```. It also removes the ability to directly target an action in the ```sampleAction``` because you have to know its index in the children beforehand.

Reply:
At the same time, if it returns the action itself, in the first line of ```sampleAction(a, s, h)``` we cannot do ```auto & states = graph_.children[a].children;``` anymore, it will have to be a search in the vector for the matching action, which is annoying. A possible solution is to return a pair<action, index> from ```runSimulation``` but that's definitely a breaking change.
Also, the ```if ( !horizon ) return 0;``` in ```runSimulation``` forces to return a number (I don't know the implications of changing that).

Reply:
Sorry for not noticing this sooner. 

I think you can return both (in the form of a tuple maybe?). If you look at policies in the POMDP they return both the action and an ID useful to sample the policy for the next horizon. I don't see this as a real problem; if it works and saves computation, it's fine (as long as it's clear what is happening).

Reply:
Yeah a pair would be great. But I think it's a breaking change. I'll find a way to minimise that.